src/Language/Haskell/Liquid/Transforms/Rewrite.hs

```diff
@@ -45,13 +45,64 @@ import qualified Data.HashMap.Strict as M
 rewriteBinds :: Config -> [CoreBind] -> [CoreBind]
 rewriteBinds cfg
   | simplifyCore cfg
-  = fmap (normalizeTuples . rewriteBindWith tidyTuples . rewriteBindWith simplifyPatTuple)
+  = fmap (normalizeTuples 
+       . rewriteBindWith undollar 
+       . rewriteBindWith tidyTuples 
+       . rewriteBindWith simplifyPatTuple)
   | otherwise
   = id
 
 simplifyCore :: Config -> Bool
 simplifyCore = not . noSimplifyCore
 
+undollar :: RewriteRule
+undollar = go 
+  where 
+    go e 
+     -- matches `$ t1 t2 t3 f a`  
+     | App e1 a  <- untick e
+     , App e2 f  <- untick e1
+     , App e3 t3 <- untick e2 
+     , Type _    <- untick t3 
+     , App e4 t2 <- untick e3 
+     , Type _    <- untick t2 
+     , App d t1  <- untick e4 
+     , Type _    <- untick t1
+     , Var v     <- untick d 
+     , show v == "GHC.Base.$" 
```

Comment:
Here's how GHC seems to do it:
https://gitlab.haskell.org/ghc/ghc/-/blob/3c0e379322965aa87b14923f6d8e1ef5cd677925/compiler/GHC/Rename/Expr.hs#L2448

I'm guessing that must be faster than calling show on every variable.

Reply:
Thanks!!!! Changed it!